traffic_ops/traffic_ops_golang/deliveryservice/request/requests.go

```diff
@@ -0,0 +1,368 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/apache/incubator-trafficcontrol/lib/go-log"
+	"github.com/apache/incubator-trafficcontrol/lib/go-tc"
+
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/auth"
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers"
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/tenant"
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+)
+
+// TODeliveryServiceRequest provides a type alias to define functions on
+type TODeliveryServiceRequest tc.DeliveryServiceRequestNullable
+
+//the refType is passed into the handlers where a copy of its type is used to decode the json.
+var refType = TODeliveryServiceRequest(tc.DeliveryServiceRequestNullable{})
+
+// GetRefType is used to decode the JSON for deliveryservice requests
+func GetRefType() *TODeliveryServiceRequest {
+	return &refType
+}
+
+//Implementation of the Identifier, Validator interface functions
+
+// GetID ...
+func (req *TODeliveryServiceRequest) GetID() int {
+	return req.ID
+}
+
+// GetAuditName ...
+func (req *TODeliveryServiceRequest) GetAuditName() string {
+	return strconv.Itoa(req.ID)
+}
+
+// GetType ...
+func (req *TODeliveryServiceRequest) GetType() string {
+	return "deliveryservice_request"
+}
+
+// SetID ...
+func (req *TODeliveryServiceRequest) SetID(i int) {
+	req.ID = i
+}
+
+// IsTenantAuthorized implements the Tenantable interface to ensure the user is authorized on the deliveryservice tenant
+func (req *TODeliveryServiceRequest) IsTenantAuthorized(user auth.CurrentUser, db *sqlx.DB) (bool, error) {
+	tenantID, err := req.getTenantID(db)
+	if err != nil {
+		log.Debugf("from getTenantID: %v", err)
+		return false, err
+	}
+	return tenant.IsResourceAuthorizedToUser(tenantID, user, db)
+}
+
+//The TODeliveryServiceRequest implementation of the Updater interface
+//all implementations of Updater should use transactions and return the proper errorType
+//ParsePQUniqueConstraintError is used to determine if a request with conflicting values exists
+//if so, it will return an errorType of DataConflict and the type should be appended to the
+//generic error message returned
+
+// Update ...
+func (req *TODeliveryServiceRequest) Update(db *sqlx.DB, user auth.CurrentUser) (error, tc.ApiErrorType) {
+	XMLID, err := req.getXMLID(db)
+	if err != nil {
+		return err, tc.SystemError
+	}
+
+	active, err := isActiveRequest(db, XMLID)
+	if err != nil {
+		return err, tc.SystemError
+	}
+	if active {
+		return errors.New("An active request exists for delivery service '" + XMLID), tc.DataConflictError
+	}
+
+	tx, err := db.Beginx()
+	defer func() {
+		if tx == nil {
+			return
+		}
+		if err == nil {
+			err = tx.Commit()
+		}
```

Comment:
On a commit error, the Update function will still return `nil` for no error. The caller of Update should probably be aware that the update failed.